addon/-private/class/modifier.ts

```diff
@@ -7,7 +7,9 @@ import { ModifierArgs } from "ember-modifier/-private/interfaces";
 // SAFETY: we're lying to the compiler here about the result of `symbol(...)`,
 // and this only holds *because* the implementation of `symbol` guarantees
 // sufficient uniqueness for this to make sense.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const DESTROYING: unique symbol = symbol("destroying") as any;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const DESTROYED: unique symbol = symbol("destroyed") as any;
```

Comment:
Could these be e.g. `export const DESTROYED = symbol(“destroyed”) as unique symbol` to avoid the linter directive comments? (Or maybe even update the return type of `symbol` with a comment on the correctness there?)

Reply:
Unfortunately, the first one doesn't work: `as unique symbol` is not allowed syntax, actually! That was the first thing I tried, but you get:

> 'unique symbol' types are not allowed here. ts(1335)

I think doing it on `symbol` is the right move, though: just make it return `any` and comment about safety there. (Especially since we'll be dropping it shortly.)